test/sample-test/run_test.sh

```diff
@@ -101,242 +103,217 @@ while [ "$1" != "" ]; do
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-local-confusion-matrix:\([a-zA-Z0-9_.-]\)\+|${LOCAL_CONFUSIONMATRIX_IMAGE}|g" xgboost-training-cm.yaml
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-local-roc:\([a-zA-Z0-9_.-]\)\+|${LOCAL_ROC_IMAGE}|g" xgboost-training-cm.yaml
+# Run the tests
+preparation ${TEST_NAME}
+
+if [[ "${TEST_NAME}" == "kubeflow_training_classification" ]]; then
+  #TODO(numerology): convert the sed commands to sed -e
+  # 's|gcr.io/ml-pipeline/|gcr.io/ml-pipeline-test/' and tag replacement. Also
+  # let the postsubmit tests refer to yaml files.
+
+  dsl-compile --py "${TEST_NAME}.py" --output "${TEST_NAME}.yaml"
+  check_result ${TEST_NAME}
+elif [[ "${TEST_NAME}" == "tfx_cab_classification" ]]; then
+  dsl-compile --py "${TEST_NAME}.py" --output "${TEST_NAME}.yaml"
+  if [[ -n "${DATAFLOW_TFT_IMAGE}" ]]; then
+    tfx_cab_classification_injection
   fi
-  cd "${TEST_DIR}"
-  python3 run_xgboost_test.py --input ${BASE_DIR}/samples/core/xgboost-spark/xgboost-training-cm.yaml --result $SAMPLE_XGBOOST_TEST_RESULT --output $SAMPLE_XGBOOST_TEST_OUTPUT --namespace ${NAMESPACE}
-
-  echo "Copy the test results to GCS ${RESULTS_GCS_DIR}/"
-  gsutil cp ${SAMPLE_XGBOOST_TEST_RESULT} ${RESULTS_GCS_DIR}/${SAMPLE_XGBOOST_TEST_RESULT}
-elif [ "$TEST_NAME" == "notebook-tfx" ]; then
-  SAMPLE_NOTEBOOK_TFX_TEST_RESULT=junit_SampleNotebookTFXOutput.xml
-  SAMPLE_NOTEBOOK_TFX_TEST_OUTPUT=${RESULTS_GCS_DIR}
-
-  # CMLE model name format: A name should start with a letter and contain only letters, numbers and underscores.
+  check_result ${TEST_NAME}
+elif [[ "${TEST_NAME}" == "xgboost_training_cm" ]]; then
+  dsl-compile --py "${TEST_NAME}.py" --output "${TEST_NAME}.yaml"
+  if [[ -n "${DATAPROC_CREATE_CLUSTER_IMAGE}" ]]; then
+    xgboost_training_cm_injection
+  fi
+  check_result ${TEST_NAME}
+elif [[ "${TEST_NAME}" == "sequential" ]]; then
+  dsl-compile --py "${TEST_NAME}.py" --output "${TEST_NAME}.yaml"
+  check_result ${TEST_NAME}
+elif [[ "${TEST_NAME}" == "condition" ]]; then
+  dsl-compile --py "${TEST_NAME}.py" --output "${TEST_NAME}.yaml"
+  check_result ${TEST_NAME}
+elif [[ "${TEST_NAME}" == "exit_handler" ]]; then
+  dsl-compile --py "${TEST_NAME}.py" --output "${TEST_NAME}.yaml"
+  check_result ${TEST_NAME}
+elif [[ "${TEST_NAME}" == "parallel_join" ]]; then
+  dsl-compile --py "${TEST_NAME}.py" --output "${TEST_NAME}.yaml"
+  check_result ${TEST_NAME}
+elif [[ "${TEST_NAME}" == "recursion" ]]; then
+  dsl-compile --py "${TEST_NAME}.py" --output "${TEST_NAME}.yaml"
+  check_result ${TEST_NAME}
+elif [[ "${TEST_NAME}" == "kubeflow_pipeline_using_TFX_OSS_components" ]]; then
+  # CMLE model name format: A name should start with a letter and contain only
+  # letters, numbers and underscores.
   DEPLOYER_MODEL=`cat /proc/sys/kernel/random/uuid`
   DEPLOYER_MODEL=Notebook_tfx_taxi_`echo ${DEPLOYER_MODEL//-/_}`
 
   cd ${BASE_DIR}/samples/core/kubeflow_pipeline_using_TFX_OSS_components
```

Comment:
Isn't this already in the sample folder after running the 'prepare' function above?

Reply:
Good catch. Thanks!